internal/viridian/api.go

```diff
@@ -112,17 +114,32 @@ func (a API) DownloadConfig(ctx context.Context, clusterID string) (path string,
 	return download(ctx, url, a.token)
 }
 
-func (a API) findClusterID(ctx context.Context, idOrName string) (string, error) {
+func (a API) FindCluster(ctx context.Context, idOrName string) (Cluster, error) {
 	clusters, err := a.ListClusters(ctx)
 	if err != nil {
-		return "", err
+		return Cluster{}, err
 	}
 	for _, c := range clusters {
 		if c.ID == idOrName || c.Name == idOrName {
-			return c.ID, nil
+			return c, nil
+		}
+	}
+	return Cluster{}, fmt.Errorf("no such cluster found: %s", idOrName)
+}
+
+func (a API) FindClusterType(ctx context.Context, name string) (ClusterType, error) {
+	clusterTypes, err := a.ListClusterTypes(ctx)
+	if err != nil {
+		return ClusterType{}, err
+	}
+	var found ClusterType
+	for _, ct := range clusterTypes {
+		if strings.ToLower(ct.Name) == strings.ToLower(name) {
```

Comment:
Original name is in upper-case, so it would be better to compare upper case strings.